### DISCUSSION TITLE
how to make `Deploy via ImageName` react to latest push in docker registry

### DISCUSSION THREAD

**Discussion by kunjee17** [caprover/caprover#1832] [Q&A] [answered]
I am using this method called `Deploy via ImageName` . It works perfectly fine for given version and also for latest tag. But if I have deployed the image using latest tag and then that image get updated. It is not re-pulling the image. 

Is there any way to make that possible? I wanted to join CI/CD with CapRover. I know direct git repo access and build on server it self but I wan't avoid server build and directly deploy docker image to save power and time on deployment server. 

Is there any options available or am I missing something?

**Comment by githubsaturn** [comment, marked as answer]
TLDR is that [Latest tag is bad](https://vsupalov.com/docker-latest-tag/#:~:text=Latest%20is%20Not%20Descriptive%20and%20Hard%20to%20Work%20With&text=You%20should%20avoid%20using%20the,and%20hard%20to%20roll%20back.)


There is no explicit way to make sure that the latest latest tag is being pulled and deployed. The only way that I know of that might work is to use the image sha256 of the image:
```
docker run ubuntu@sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2
```

**Reply by kunjee17** [reply]
Thanks @githubsaturn for answer. Basically I wanted that CI/ CD experience but with docker images.
